streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -917,8 +922,34 @@ public void punctuate(final ProcessorNode<?, ?, ?, ?> node,
             maybeMeasureLatency(() -> node.punctuate(timestamp, punctuator), time, punctuateLatencySensor);
         } catch (final StreamsException e) {
             throw e;
-        } catch (final RuntimeException e) {
-            throw new StreamsException(String.format("%sException caught while punctuating processor '%s'", logPrefix, node.name()), e);
+        } catch (final Exception e) {
```

Comment:
Why do we catch `Exception` here, but not `RuntimeException`? If no checked exception is declared, it seem not to be "correct"?

Reply:
@mjsax For this reason: https://github.com/apache/kafka/pull/16300#discussion_r1687844510

`Exception` is caught in `ProcessorNode#process`, `RecordCollectorImpl#handleException` and `RecordDeserializer#deserialize` so we keep `Exception` here to be consistent

Reply:
Thanks. I disagree to @cadonna conclusion -- we should rather catch `RuntimeException` everywhere (consistency yes, but the other way around).

I'll cover this on the other PR I did open.